controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -126,24 +120,79 @@ protected void startUp() throws InterruptedException, Exception {
                 }
             };
 
-            taskSweeper.sweepOrphanedTasks(processes);
+            sweepEventProcessorReaders(processes);
+            sweepTransactions(processes);
+            sweepTasks(processes);
+
+            log.info("Controller cluster listener startUp complete");
+        } finally {
+            LoggerHelpers.traceLeave(log, objectId, "startUp", traceId);
+        }
+    }
+
+    private void handleHostRemoved(Host host) {
+        taskSweeper.sweepOrphanedTasks(host.getHostId());
+
+        eventProcessorsOpt.ifPresent(controllerEventProcessors -> {
+            Supplier<CompletableFuture<Void>> futureSupplier = () -> RetryHelper.withIndefiniteRetriesAsync(() -> {
```

Comment:
This needs to be retried forever because if we fail in sweeping then we lose the opportunity to pick the left over work from another controller instance. 
So we need to make sure that transfer of ownership of the work happens conclusively